lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for Yoda conditions where the two parts of an expression
+      # are reversed from the typical order in a conditional statement.
```

Comment:
I took a closer look at the wikipedia article linked to on #4145 to get some ideas, and I realized this description is copied-and-pasted from there.  So whatever wording we choose here should probably be "back-ported" to the wikipedia article.

So I'll start by quoting from the end of that paragraph:
```
Yoda, who spoke English in a non-standard syntax.
```

and from later in the article (in the Criticism section):

```
Critics of Yoda conditions see the lack of readability as a disadvantage that
outweighs the benefits described above. 
```

With that in mind, I propose the following:
```
This cop checks for Yoda conditions, i.e. comparison operations where
readability is reduced because the operands are not ordered the same
way as they would be ordered in spoken English.
```

But feel free to change "spoken English" to "spoken language" if to your knowledge this is true in all other relevant languages here as well.